### ISSUE TITLE
model.isDirty() reactivity for VueJS

### ISSUE THREAD

**Issue by elDub** [graphiti-api/spraypaint.js#13]
I'd like to be able to use the `model.isDirty()` method within VueJS computed values for immediate DOM updates when a model becomes dirty. I believe that this currently does not work because of the way VueJS observes referenced objects within a computed method for property access on first run and subsequently assigns getter/setter methods with which it can determine that a change has been made and therefore trigger a regeneration of the computed value.

My first thought was to touch/read all model attribute methods within `isDirty()`, but because of the way it (rightfully) short-circuits checking _all_ objects when the first dirty object is found, it wouldn't trigger reactivity in all places. Perhaps a method on the models that could be called in tandem within the computed value?

Or perhaps a better implementation if my need is valid for this project?

Here is a sample of what I'm currently thinking of doing.

HTML:
```pug
  div.alert(v-if="assetChanged") You have unsaved data!
```

VueJS component:
```coffeescript
computed:
  assetChanged: ->
    if @.asset
      relationships = ['relation1', 'relation2': 'child-relation']
      @.asset.establishReactivity(relationships)
      @.asset.isDirty(relationships)
```

**Comment by richmolj** [comment]
@elDub if you wouldn't mind a little test, not sure this is the right way to go but curious if it works. If we instantiated attributes with all the keys and null values, instead of `{}` [here](https://github.com/graphiti-api/spraypaint.js/blob/master/src/model.ts#L423), does it fix the issue? Maybe not for relationships, but at least for the first part?

**Comment by elDub** [comment]
I don't think that populating `model._attributes` on instantiation will do the trick. In order to trigger a computed value to re-evaluate, a setter must be accessed for which a related getter was during the first run of the computed value. (At least that is how I understand they work.) This means that for the first run of the computed value, the code that is important must check all spots that could be altered. It is also the reason that I don't expect `isDirty` to be the method to do the trick since it _should_ stop processing once it finds the first change, and therefore the first run of it would not properly register _all_ spots that need to be observed.

This is my hack-for-the-moment to get this to work. I had to copy your `_parse*` methods (not shown) as they are not accessible from outside the library to get this to work.

**application_record.coffee**
```coffee
ApplicationRecord = SpraypaintBase.extend
  methods:
    establishReactivity: (relations) ->
      # Accessing the properties establishes reactivity
      @[property] for property in Object.keys(@)

      relationshipHash = _parseIncludeArgs relations
      for relationship of relationshipHash
        if Array.isArray @[relationship]
          for child in @[relationship]
            child.establishReactivity relationshipHash[relationship]
        else
          @[relationship].establishReactivity relationshipHash[relationship]
```

**VueJS component fragment**
```coffee
  export default
    computed:
      assetChanged: ->
        if @.asset instanceof Asset
          relations = ["images", "documents", "expenses": "category"]
          @.asset.establishReactivity(relations)
          @.asset.isDirty(relations)
```

A side benefit of declaring which models are reactive _within the computed value code_ is that I can have multiple computed values that each may take into account the root object being dirty, but different relationships.
